lib/iris/pandas.py

```diff
@@ -518,88 +518,298 @@ def _get_base(array):
+    419900  2010-09-16 00:00:00  4.444450  356.666656  ...                 356844.0                0  m01s00i024
+    419901  2010-09-16 00:00:00  4.444450  357.500000  ...                 356844.0                0  m01s00i024
+    419902  2010-09-16 00:00:00  4.444450  358.333313  ...                 356844.0                0  m01s00i024
+    419903  2010-09-16 00:00:00  4.444450  359.166656  ...                 356844.0                0  m01s00i024
+
+    [419904 rows x 7 columns]
+
+    To retrieve a :class:`~pandas.Series` from `df` :class:`~pandas.DataFrame`, subselect a column:
+
+    >>> df['surface_temperature']
+    0         301.659271
+    1         301.785004
+    2         301.820984
+    3         301.865234
+    4         301.926819
+                ...
+    419899    298.779938
+    419900    298.913147
+    419901           NaN
+    419902           NaN
+    419903    298.995148
+    Name: surface_temperature, Length: 419904, dtype: float32
 
     """
-    data = cube.data
+    # Checks
+    if not isinstance(cube, iris.cube.Cube):
+        raise ValueError("Input must be an iris.cube.Cube instance")
+    if add_global_attributes:
+        global_attribute_names = list(cube.attributes.keys())
+        for global_attribute in add_global_attributes:
+            if (
+                global_attribute not in global_attribute_names
+            ):  # Check global attribute exists
+                raise ValueError(
+                    f'"{global_attribute}" not found in cube attributes'
+                )
+
+    if copy:
+        data = cube.data.copy()
+    else:
+        data = cube.data
     if ma.isMaskedArray(data):
-        if not copy:
-            raise ValueError("Masked arrays must always be copied.")
         data = data.astype("f").filled(np.nan)
-    elif copy:
-        data = data.copy()
 
-    index = columns = None
-    if cube.coords(dimensions=[0]):
-        index = _as_pandas_coord(cube.coord(dimensions=[0]))
-    if cube.coords(dimensions=[1]):
-        columns = _as_pandas_coord(cube.coord(dimensions=[1]))
+    # Extract dim coord information
+    dim_coord_list = [
+        [
+            cube.coords(dimensions=n, dim_coords=True)[0].name(),
+            _as_pandas_coord(cube.coords(dimensions=n, dim_coords=True)[0]),
+        ]
+        if cube.coords(dimensions=n, dim_coords=True)
+        else ["dim" + str(n), range(cube.shape[n])]
+        for n in range(cube.ndim)
+    ]
+    # Extract separate lists for dim names and dim values
+    coord_names, coords = list(zip(*dim_coord_list))
+    # Make base DataFrame
+    index = pandas.MultiIndex.from_product(coords, names=coord_names)
+    data_frame = pandas.DataFrame(
+        data.ravel(), columns=[cube.name()], index=index
+    )
 
-    data_frame = pandas.DataFrame(data, index, columns)
-    if not copy:
-        _assert_shared(data, data_frame)
+    # Add AuxCoord & scalar coordinate information
```

Comment:
To align with the features of `iris.pandas.as_cubes()`: can the code for `AuxCoord` columns be generalised to do `CellMeasure`s and `AncillaryVariable`s, too? These all come from the same parent class so I would hope there would only be a few pieces to change.

Reply:
Done, but now added additional kwargs for syntactic consistency with `as_cubes`